lib/mappers/esdn_mapper.py

```diff
@@ -0,0 +1,331 @@
+                        _dict["object"] = url.get("#text")
+                except:
+                    pass
+
+            self.mapped_data.update(self.clean_dict(_dict))
+
+    #
+    def map_language(self):
+        prop = self.root_key + "language/languageTerm"
+        lang = None
+        if exists(self.provider_data, prop):
+            lang = getprop(self.provider_data, prop)
+
+            if isinstance(lang, dict):
+                try:
+                    lang = lang.get("#text")
+                except:
+                    pass
+
+        if lang:
+            self.update_source_resource({"language": lang})
+
+    def map_relation(self):
+        prop = self.root_key + "relatedItem/titleInfo/title"
+
+        if exists(self.provider_data, prop):
+            self.update_source_resource({"relation":
+                                         getprop(self.provider_data, prop)})
+
+    #
+    def map_rights(self):
+        prop = self.root_key + "accessCondition"
+        rights = None
+        if exists(self.provider_data, prop):
+            rights = getprop(self.provider_data, prop)
+            if isinstance(rights, dict):
+                rights = rights.get("#text")
+
+        if rights:
+            self.update_source_resource({"rights": rights })
+
+    #
+    def map_subject_and_spatial_and_temporal(self):
+        prop = self.root_key + "subject"
+        _dict = {
+            "subject": [],
+            "spatial": [],
+            "temporal": []
+        }
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                try:
+                    if "geographic" in s and s.get("geographic"):
+                        _dict["spatial"].append(s.get("geographic"))
+                    elif "topic" in s and s.get("topic"):
+                        _dict["subject"].append(s.get("topic"))
+                    elif "temporal" in s and s.get("temporal"):
+                        _dict["temporal"].append(s.get("temporal"))
+                except:
+                    logger.error("Error when trying to map geo/topic/temporal")
+            for k in _dict.keys():
+                _dict[k] = self.unnest_list(_dict.get(k), [])
+
+            self.update_source_resource(self.clean_dict(_dict))
+
+    #
+    def map_title(self):
+        prop = self.root_key + "titleInfo/title"
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                if isinstance(s, unicode) or not s.keys:
+                    self.update_source_resource({"title": getprop(self.provider_data, prop)})
+
+    # <typeofresource> when the value matches DCMIType
+    def map_type(self):
+        prop = self.root_key + "typeOfResource"
+
+        if exists(self.provider_data, prop):
+            self.update_source_resource({"type": getprop(self.provider_data, prop)})
+
+    def unnest_list(self, nested_array, array=[]):
```

Comment:
Yeah, this was frustrating to write; seemed like there should be something much more elegant